File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    num_list = [i for i in range(0,n+1)]
    
    for i in range(1,n+1):
        x=0
        for j in range(i,n+1):
            x += j
            if x == n:
                answer += 1
                break
            elif x > n:
                break
    
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for (int i = 1; i< n+1;i++) {
            int x = i;
            for (int j = i + 1;j < n + 1;j++) {
                x += j;
                if (x == n) {
                    answer++;
                    break;
                } else if (x > n) {
                    break;
                }
            }
        }
        answer++;
        
        return answer;
    }
}